DataBiosphere/Saturn-UI
# [DC-783]Call service endpoint for creating a Snapshot Request

### Jira Ticket: https://broadworkbench.atlassian.net/browse/DC-783


### NEXT STEPS: 
Currently, this passes in a dummy value for id in the valueSets. This is a field that should be removed in TDR (represented by SnapshotBuilderFeatureValueGroup). Then, removed here as well.
### UPDATE
The dummy value has been removed in both the UI and TDR (See DC-783 in TDR). This branch now works with the TDR branch DC-783 and not the current dev backend. 

## Summary of changes:

### What
- This adds a call to the endpoint for createSnapshotRequest.
-  This requires that datasetId be passed into the request access modal. 
- This requires conversion between the data types used on the UI and the API. So new data types and conversions are added to the ajax/DatasetBuilder page.

### Why
- To create a request for access for a snapshot when a new snapshot request is submitted.

### Testing strategy
- [ ] When running the UI locally, go through the user flow for requesting a new snapshot and when submitting verify in the network tab that the request was made with a 200 response.
In order to test these changes I needed to connect the UI to the locally running TDR. These are the steps I followed and a video of the result. 
1. Run local TDR on branch DC-783 
2. In UI dev.json, point to http://localhost:8080/ for TDR
3. Create new dataset using AzureY1Demo.ipynb
Create billing profile 
Use gcp everywhere
When authenticating with jwt, use idp_access_token
4. Use http://localhost:8080/swagger-ui.html#/datasets/updateDatasetSnapshotBuilderSettings to add snapshot settings 
Print to json TDR’s SnapshotBuilderTestData.SETTINGS 
5. Run UI locally 
6. Open http://localhost:3000/#library/builder/2681e4a0-2f08-438d-8cfa-a4bffe148e58/build in safari, disable cross-origin restrictions
7. Go through user flow to create a snapshot request!

### Visual Aids 

https://github.com/DataBiosphere/terra-ui/assets/55256690/cee1081d-22dd-4620-98be-323f9876e840
